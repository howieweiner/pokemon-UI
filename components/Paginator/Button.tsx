import React from 'react'
import classNames from 'classnames'

type Props = {
  label: string
  disabled: boolean
  onSelect: () => void
}

const Button: React.FC<Props> = ({ label, disabled, onSelect }) => {
  const styles = classNames(
    'flex items-center justify-center px-2 py-2 mx-1 text-sm text-center w-20 rounded-md bg-white',
    {
      'text-gray-700 transition-colors duration-200 transform dark:bg-gray-800 dark:text-gray-200 hover:bg-gray-600 dark:hover:bg-gray-500 hover:text-white dark:hover:text-gray-200':
        !disabled,
      'text-gray-400 cursor-not-allowed dark:bg-gray-800 dark:text-gray-600': disabled,
    }
  )

  const onSelected = (): void => {
    onSelect()
  }

  return (
    <button className={styles} onClick={onSelected} onKeyPress={onSelected} data-testid="paginator-button">
      {label}
    </button>
  )
}

export default Button
