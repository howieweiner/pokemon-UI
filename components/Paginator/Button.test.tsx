import React from 'react'
import Button from './Button'
import { fireEvent, render } from '@testing-library/react'

describe('<Button />', () => {
  it('displays the label text', async () => {
    const { findByText } = render(<Button label="Press Me!" disabled={true} onSelect={jest.fn()} />)
    await findByText('Press Me!')
  })

  it('fires the select event when the button is clicked', () => {
    const mockEvent = jest.fn()
    const { getByTestId } = render(<Button label="Press Me!" disabled={true} onSelect={mockEvent} />)
    fireEvent.click(getByTestId('paginator-button'))
    expect(mockEvent).toHaveBeenCalled()
  })

  it('fires the select event when the button is selected using keyboard', () => {
    const mockEvent = jest.fn()
    const { getByTestId } = render(<Button label="Press Me!" disabled={true} onSelect={mockEvent} />)
    fireEvent.keyPress(getByTestId('paginator-button'), { key: 'Enter', keyCode: 13 })
    expect(mockEvent).toHaveBeenCalled()
  })

  it('renders the button as disabled', () => {
    const { container } = render(<Button label="Press Me!" disabled={true} onSelect={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })

  it('renders the button as enabled', () => {
    const { container } = render(<Button label="Press Me!" disabled={false} onSelect={jest.fn()} />)
    expect(container).toMatchSnapshot()
  })
})
