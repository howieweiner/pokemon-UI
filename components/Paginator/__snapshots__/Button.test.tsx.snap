// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders the button as disabled 1`] = `
<div>
  <a
    class="flex items-center justify-center px-2 py-2 mx-1 text-sm text-center w-20 rounded-md bg-white text-gray-400 cursor-not-allowed dark:bg-gray-800 dark:text-gray-600"
    data-testid="paginator-button"
    href="#"
  >
    Press Me!
  </a>
</div>
`;

exports[`<Button /> renders the button as enabled 1`] = `
<div>
  <a
    class="flex items-center justify-center px-2 py-2 mx-1 text-sm text-center w-20 rounded-md bg-white text-gray-700 transition-colors duration-200 transform dark:bg-gray-800 dark:text-gray-200 hover:bg-gray-600 dark:hover:bg-gray-500 hover:text-white dark:hover:text-gray-200"
    data-testid="paginator-button"
    href="#"
  >
    Press Me!
  </a>
</div>
`;
